src/frontend/sharedComponents/PracticeMode.tsx

```diff
@@ -29,38 +29,37 @@ export const PracticeMode = ({
 }: PracticeModeProps) => {
   const [config] = useContext(ConfigContext);
 
-  if (enabled && isInPracticeMode(config)) {
-    return (
-      <View style={styles.container}>
-        <View style={styles.contentContainer}>{children}</View>
+  const showPracticeModeUi = enabled && isInPracticeMode(config);
 
-        {!hideBar && (
-          <View style={styles.bottomBar}>
-            <MaterialCommunityIcons
-              name="lightbulb-on"
-              color={WHITE}
-              size={28}
-            />
-            <Text style={styles.text}>
-              <FormattedMessage {...m.title} />
-            </Text>
-          </View>
-        )}
-      </View>
-    );
-  }
+  return (
+    <View
+      style={[
+        styles.flexContainer,
+        showPracticeModeUi && styles.practiceModeContainer,
+      ]}
+    >
+      <View style={styles.flexContainer}>{children}</View>
 
-  //If practice mode is NOT enabled, the user is just returned the children
-  return <React.Fragment>{children}</React.Fragment>;
+      {showPracticeModeUi && !hideBar && (
+        <View style={styles.bottomBar}>
+          <MaterialCommunityIcons name="lightbulb-on" color={WHITE} size={28} />
+          <Text style={styles.text}>
+            <FormattedMessage {...m.title} />
+          </Text>
+        </View>
+      )}
+    </View>
+  );
```

Comment:
Instead of relying on conditional returns to hide practice mode UI, we conditionally style a wrapping container. This prevents massive re-renders with the relatively minor trade-off of always having the wrapper present